test/rubygems/test_gem_platform.rb

```diff
@@ -261,6 +264,42 @@ def test_nil_cpu_arch_is_treated_as_universal
     assert((with_x86_arch === with_nil_arch), 'x86 =~ nil')
   end
 
+  def test_nil_version_is_treated_as_any_version
+    x86_darwin_8 = Gem::Platform.new 'i686-darwin8.0'
+    x86_darwin_nil = Gem::Platform.new 'i686-darwin'
+
+    assert((x86_darwin_8 === x86_darwin_nil), '8.0 =~ nil')
+    assert((x86_darwin_nil === x86_darwin_8), 'nil =~ 8.0')
+  end
+
+  def test_nil_version_is_stricter_for_linux_os
+    x86_linux = Gem::Platform.new 'i686-linux'
+    x86_linux_gnu = Gem::Platform.new 'i686-linux-gnu'
+    x86_linux_musl = Gem::Platform.new 'i686-linux-musl'
+    x86_linux_uclibc = Gem::Platform.new 'i686-linux-uclibc'
+
+    # a naked linux runtime is implicit gnu, as it represents the common glibc-linked runtime
+    assert(x86_linux === x86_linux_gnu, 'linux =~ linux-gnu')
+    assert(x86_linux_gnu === x86_linux, 'linux-gnu =~ linux')
+
+    # musl and explicit gnu should differ
+    # refute(x86_linux_gnu === x86_linux_musl, 'linux-gnu =~ linux-musl')
+    # refute(x86_linux_musl === x86_linux_gnu, 'linux-musl =~ linux-gnu')
```

Comment:
It might be worth thinking about introducing `linux-gnu` someday, for those good willed maintainers that know their binary is definitely not musl compatible but with no `linux-musl` published (yet, or ever).

Reply:
I did introduce checks for this but I think we may be getting ahead of ourselves, I'm not fully sure whether we should include it with this patch or not.